src/consumer/topic.rs

```diff
@@ -95,45 +96,91 @@ impl<T: DeserializeMessage, Exe: Executor> TopicConsumer<T, Exe> {
+
+                                current_retries += 1;
+                                client
+                                    .executor
+                                    .delay(operation_retry_options.retry_delay)
+                                    .await;
+
+                                // we need to look up again the topic's address
+                                let prev = addr;
+                                addr = client.lookup_topic(&topic).await?;
+                                if prev != addr {
+                                    info!(
+                                        "topic {} moved: previous = {:?}, new = {:?}",
+                                        topic, prev, addr
+                                    );
+                                }
+
+                                connection = client.manager.get_connection(&addr).await?;
+                                continue;
+                            }
+                            _ => {
+                                error!("subscribe({}) reached max retries", topic);
+
+                                return Err(ConnectionError::PulsarError(
+                                    Some(proto::ServerError::ServiceNotReady),
+                                    text,
+                                )
+                                .into());
+                            }
+                        }
+                    }
+                }
+                Err(ConnectionError::Io(e))
+                    if matches!(
+                        e.kind(),
+                        ErrorKind::ConnectionReset
+                            | ErrorKind::ConnectionAborted
+                            | ErrorKind::NotConnected
+                            | ErrorKind::BrokenPipe
+                            | ErrorKind::TimedOut
+                            | ErrorKind::Interrupted
+                            | ErrorKind::UnexpectedEof
+                    ) =>
+                {
+                    match operation_retry_options.max_retries {
+                        Some(max_retries) if current_retries < max_retries => {
+                            error!(
+                                    "create consumer( {} {}, retrying request after {}ms (max_retries = {:?})",
+                                    topic, e.kind().to_string(), operation_retry_options.retry_delay.as_millis(),
+                                    operation_retry_options.max_retries
+                                );
+
+                            current_retries += 1;
+                            client
+                                .executor
+                                .delay(operation_retry_options.retry_delay)
+                                .await;
+
+                            let addr = client.lookup_topic(&topic).await?;
+                            connection = client.manager.get_connection(&addr).await?;
+
+                            continue;
```

Comment:
Why do we use different re-lookup logic here compare to the snippet above?

```rust
                            // we need to look up again the topic's address
                            let prev = addr;
                            addr = client.lookup_topic(&topic).await?;
                            if prev != addr {
                                info!(
                                    "topic {} moved: previous = {:?}, new = {:?}",
                                    topic, prev, addr
                                );
                            }

                            connection = client.manager.get_connection(&addr).await?;
                            continue;
```

Reply:
@CleverAkanoa ping :D